driver/src/orderbook/streamed/state.rs

```diff
@@ -652,4 +691,93 @@ mod tests {
             2
         );
     }
+
+    #[test]
+    fn orderbook_batch_id() {
+        let mut state = state_with_fee();
+        let event = Deposit {
+            user: address(3),
+            token: address(0),
+            amount: 1.into(),
+            batch_id: 0,
+        };
+        state = state.apply_event(&Event::Deposit(event), 0).unwrap();
+        let balance = state.orderbook(Batch::Current).unwrap().0.next().unwrap().1;
+        assert_eq!(balance, U256::zero());
+        assert!(state.orderbook(Batch::Future(0)).is_err());
```

Comment:
Is this an error because there is no way 0 is in the future?

Reply:
It was an error (just changed it) because batch id 0 is not actually in the future and the code creating the orderbook checked that.